### ISSUE TITLE
Ransac code executing  

### ISSUE THREAD

**Issue by AymenFJA** [iceberg-project/ASIFT#27]
Hello @mmacferrin ,

when i try to execute the "ransac_filter.py" using  `python3` and `opencv 3.2` i am getting : 

``` File "ransac_filter.py", line 128
    print numpy.count_nonzero(mask), "of", len(mask), "matches selected."
    SyntaxError: invalid syntax ``` 
``` 
When  i switched to ``` python2 ```  and opencv 2.4 i got: 

``` File "ransac_filter.py", line 21, in <module>
    from csv_writer import write_csv
    ImportError: No module named csv_writer ```

can you help me with this part ?

**Comment by mmacferrin** [comment]
Hi @AymenFJA, the code is written in Python 2.7, so it won't work running it with Python 3 (mostly involving the print statments which in Python 3 are required to have parentheses ["print('Hello world.')"]). A conversion can easily be done using the python "2to3" package if we wanted to use Python 3.

Using Python 2, with the "no module named csv_writer", are the directories in lower-case? I'd changed the code to use all-lowercase names, per the request of ticket #17, but if the directories aren't named accordingly then perhaps it isn't finding/including that directory.

csv_writer.py is in the "utilities" directory. Lines 10-17 of "ransac_filter.py" import that directory into the system path to use modules there, so perhaps the case mismatch is causing the error. I thought I had renamed the directories to all-lower-case, but I'm seeing they're still upper case in my local git directory so perhaps that didn't happen (if so, my apologies!). If the names of the sub-directories are in fact in upper-case, rename them to lower-case and see if that fixes it. I will make sure they're lower-case in all future commits. If that doesn't fix it let me know and I'll see what's up.

**Comment by AymenFJA** [comment]
@mmacferrin :

Thanks for explaining these things to me. We have an open pull request waiting to be approved and I don't think it would be unless the code is actually working. I think we need to fix the lowercase uppercase and I can help you with that, I will go through your code again and I will keep updating this ticket until the code is working as it's supposed to be.

**Comment by iparask** [comment]
Unless the code runs correctly pull-request #26 will remain open.

**Comment by mmacferrin** [comment]
Yes. I'm aware the code needs to work. Thanks, Ioannis.

After a bit of searching, I've determined Git is case-insensitive, which is causing the issues here. When I renamed directory structures to lower-case per your previous ticket request, the Git commit did not update folder names. They're lower-case on my machine but still upper-case on the online repository, and it says they're fully synced. Which means when you run the code in a case-senstive Linux environment, it broke because folders weren't being imported correctly. I will work around this by renaming the folders briefly, committing, then renaming them again and committing again. It's a workaround since git doesn't recognize directory renames (involving only capitalization) as actual changes. Expect a couple back-to-back commits here, momentarily.

**Comment by mmacferrin** [comment]
Commits done. Feel free to check the code to see if this bug is fixed for you now.

**Comment by iparask** [comment]
Hello @mmacferrin,

You develop in Windows, right? I've faced that issue as well. Fortunately or unfortunately, depends how you see it, all HPCs are Unix OS based. As a result, we have to comply with that. Also, it helps with consistency.

I will leave the code review to @AymenFJA, since he knows better the use case than me.

**Comment by AymenFJA** [comment]
we are getting an error  :
```
H, mask = cv2.findHomography(keys1_ndarray, keys2_ndarray, method=cv2.RANSAC, ransacReprojThreshold=ransac_threshold, maxIters=2000)
TypeError: 'maxIters' is an invalid keyword argument for this function 
```

**Comment by AymenFJA** [comment]
fixed now, waiting for the pull request #29 to be reviewed and accepted.

**Comment by AymenFJA** [comment]
closing this issue. Pull request #29 was approved by @mmacferrin and @iparask.
